src/retry.rs

```diff
@@ -0,0 +1,25 @@
+use tokio_retry::strategy::{FibonacciBackoff, jitter};
+use std::future::Future;
+use std::{
+  iter::{Take, Map},
+  time::Duration
+};
+
+fn retry_strategy(ms: u64, attempts: usize) -> Take<Map<FibonacciBackoff, fn(Duration) -> Duration>>{
+  FibonacciBackoff::from_millis(ms)
+    .map(jitter as fn(Duration) -> Duration)
+    .take(attempts)
+}
+
+
+pub async fn with_retry<A, F, R, E>(ms: Option<u64>, attempts: Option<usize>, action: A) -> Result<R, E>
```

Comment:
```suggestion
pub async fn with_retry<A, F, R, E>(base_delay: Duration, attempts: usize, action: A) -> Result<R, E>
```

Let's put the defaults into the client module as constants and make them configurable using `LogglyClientBuilder`.